crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,430 @@
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// The space around and inside of a UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Spacing {
+    /// The space around the outside of the UI element
+    pub margin: UiRect,
+    /// The space around the inside of the UI element
+    pub padding: UiRect,
+    /// The space around the outside of the UI element that can be colored to create a visible border
+    pub border: UiRect,
+}
+
+impl Spacing {
+    pub const DEFAULT: Spacing = Spacing {
+        margin: UiRect::DEFAULT,
+        padding: UiRect::DEFAULT,
+        border: UiRect::DEFAULT,
+    };
+
+    /// Adds an automatic margin on all sides.
+    ///
+    /// When the element is explicitly sized, this will (hopefully) cause the element
+    /// to be vertically and horizontally centered relative to its parent.
+    pub const AUTO_MARGIN: Spacing = Spacing::margin(UiRect {
+        left: Val::Auto,
+        right: Val::Auto,
+        bottom: Val::Auto,
+        top: Val::Auto,
+    });
+
+    /// Sets only the margin
+    pub const fn margin(rect: UiRect) -> Spacing {
+        Spacing {
+            margin: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the margin on all sides to `val`
+    pub const fn margin_all(val: Val) -> Spacing {
+        Spacing {
+            margin: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn padding(rect: UiRect) -> Spacing {
+        Spacing {
+            padding: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the padding on all sides to `val`
+    pub const fn padding_all(val: Val) -> Spacing {
+        Spacing {
+            padding: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn border(rect: UiRect) -> Spacing {
+        Spacing {
+            border: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the border on all sides to `val`
+    pub const fn border_all(val: Val) -> Spacing {
+        Spacing {
+            border: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// Whether to show or hide overflowing items
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Reflect, Serialize, Deserialize,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Overflow {
+    /// Show overflowing items
+    #[default]
+    Visible,
+    /// Hide overflowing items
```

Comment:
Is this "hide" or "clip"?

Reply:
Should say clip indeed. They get clipped, not hidden.

Reply:
```suggestion
    /// Clip overflowing items
```

Reply:
Or perhaps we could directly use `Clipped`? Yet another difference with CSS but it'd be okay as it's an enum.